### DISCUSSION TITLE
Deploying a Dotnet Web App

### DISCUSSION THREAD

**Discussion by sreekanthedayar** [1Panel-dev/1Panel#7497] [Q&A]
Hi, I saw dotnet runtime option, i tried to deploy a dotnet web api with runtime, but its getting errored. Do i need to deploy in containerised mode or can i upload the published files with target runtime?

**Comment by zhengkunwang223** [comment]
> Hi, I saw dotnet runtime option, i tried to deploy a dotnet web api with runtime, but its getting errored. Do i need to deploy in containerised mode or can i upload the published files with target runtime?

You need to build a .dll file yourself first, and then run it using 1Panel's runtime. This is our current logic. Perhaps we could support building and running directly? Looking forward to your feedback.

**Reply by sreekanthedayar** [reply]
I had a working published files as framework dependent mode, with startup dll. Which i currently deploy manually on ubuntu with target to dotnet runtime. I uploaded same working published files and tried to setup. It always ended up in error. If you want i can try to reproduce it. I also did't saw the dotnet runtime installed.

**Reply by zhengkunwang223** [reply]
> I had a working published files as framework dependent mode, with startup dll. Which i currently deploy manually on ubuntu with target to dotnet runtime. I uploaded same working published files and tried to setup. It always ended up in error. If you want i can try to reproduce it. I also did't saw the dotnet runtime installed.

You can show the logs and errors
It would be best if you could provide screenshots

**Reply by sreekanthedayar** [reply]
This is the step that i did
1. Installed panel on Ubuntu 22.04
2. Installed MySQL
3. Uploaded my published API files (which is already running on another machine) to www/wwwwroot/apifolder 
4. Set permission to 755 for the above location
5. Create .net runtime with startup dll mentioned 

its showing stopped
[2Kapi-dev2  | /run.sh: line 5: ClapupApis1_2.dll: command not found
[2Kapi-dev2  | /run.sh: line 5: ClapupApis1_2.dll: command not found
[2Kapi-dev2  | /run.sh: line 5: ClapupApis1_2.dll: command not found
[2Kapi-dev2  | /run.sh: line 5: ClapupApis1_2.dll: command not found
[2Kapi-dev2  | /run.sh: line 5: ClapupApis1_2.dll: command not found
[2Kapi-dev2  | /run.sh: line 5: ClapupApis1_2.dll: command not found

What i noticed 
1. When i tried to install dotnet from appstore, nothing happened other than redirecting me to website/runtime location in the panel without nothing installed. 

Another point, we use systemd and systemdfile to add environment variables, connection string to db etc. in this case wheer we can add those variables?

**Reply by sreekanthedayar** [reply]
![image](https://github.com/user-attachments/assets/3f5dd201-0bb1-4e2c-8759-e42cea833dba)
![image](https://github.com/user-attachments/assets/34f02ea2-4d18-4926-aaa8-0ac1fe6afc72)

**Reply by zhengkunwang223** [reply]
> ![image](https://private-user-images.githubusercontent.com/43740828/397904118-3f5dd201-0bb1-4e2c-8759-e42cea833dba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3NjM5MjgsIm5iZiI6MTczNDc2MzYyOCwicGF0aCI6Ii80Mzc0MDgyOC8zOTc5MDQxMTgtM2Y1ZGQyMDEtMGJiMS00ZTJjLTg3NTktZTQyY2VhODMzZGJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIxVDA2NDcwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM3NjEwYzNhN2RlZTMyMjcwNmMxNzlhMTI5OWRhMTljMGViNTJjZmVjMzljYjI3YTk4MDM3MTI1OWRlMjU3NTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.O3Ta_-ArXWur4zB4rrnX3BDesrOzLiT5AiWC2Rd3nuw) ![image](https://private-user-images.githubusercontent.com/43740828/397904128-34f02ea2-4d18-4926-aaa8-0ac1fe6afc72.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3NjM5MjgsIm5iZiI6MTczNDc2MzYyOCwicGF0aCI6Ii80Mzc0MDgyOC8zOTc5MDQxMjgtMzRmMDJlYTItNGQxOC00OTI2LWFhYTgtMGFjMWZlNmFmYzcyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIxVDA2NDcwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZmYWViYzdhNDUwNDQ0NzVhZTI0MmUzYzQ2ZjA5MDFlZjE5NTIxYmMwZjgwMTk2OWMxNzljNmUzYzVhYjRjYjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.b3iqcpDYB8gwZNCRL92aOfanqyBMdI6wyyUdVu5KG3M)

You should change the run script  , set donet xxx.dll

**Reply by sreekanthedayar** [reply]
Okay let me try. Now where can i add my environment variables? i had done and it's showing running at port 5000. Enabled extrenal access. Tried to setup reverse proxy and website with runtime. Both end up in error, when i checked domain 

![image](https://github.com/user-attachments/assets/12e4aae6-5c41-4e08-be00-2409ae3d1d3a)

**Reply by zhengkunwang223** [reply]
> Okay let me try. Now where can i add my environment variables? i had done and it's showing running at port 5000. Enabled extrenal access. Tried to setup reverse proxy and website with runtime. Both end up in error, when i checked domain
> 
> ![image](https://private-user-images.githubusercontent.com/43740828/397911811-12e4aae6-5c41-4e08-be00-2409ae3d1d3a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3NjkyODYsIm5iZiI6MTczNDc2ODk4NiwicGF0aCI6Ii80Mzc0MDgyOC8zOTc5MTE4MTEtMTJlNGFhZTYtNWM0MS00ZTA4LWJlMDAtMjQwOWFlM2QxZDNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIxVDA4MTYyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1OWUzZjg1MmUwNGI1NjI2ODVhZTNkYjdjMTlhYTg2OThiZDIzMDkwZDRmYmFjYTJjNTg0ZTFhNjVmMGRlOTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.rL8zHXDGuOeZ1fS-uc4shrOd7WfJixqd8PkDyfP32Wc)

Currently, we do not support setting environment variables on the page. This feature will be implemented in version 2. You can navigate to the directory:  

```bash
cd /opt/1panel/runtime/dotnet/<your dotnet runtime name>
```  

Then modify the `docker-compose.yml` file and add the `environment` field.

**Reply by sreekanthedayar** [reply]
Still runtime is running, but iam unable to proxy it using reverse proxy option. Is this normal?

**Reply by zhengkunwang223** [reply]
> Still runtime is running, but iam unable to proxy it using reverse proxy option. Is this normal?

If it is a web application, you can run the following command on a Linux server:  

```bash
curl http://127.0.0.1:<your runtime port>
```  

Check if there is a response to confirm whether it is running properly.
